app/forms/keywords_form.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_reader :file, :keyword_ids
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(file)
+    @file = file
+
+    return false unless valid?
+
+    begin
+      keyword_records = parse_keywords.map { |keyword| keyword_record(keyword) }
+      # rubocop:disable Rails::SkipsModelValidations
+      @keyword_ids = Keyword.insert_all(keyword_records).map { |hash| hash['id'] }
+      # rubocop:enable Rails::SkipsModelValidations
+    rescue ActiveRecord::ActiveRecordError
+      errors.add(:base, I18n.t('keywords.upload.invalid_file'))
+    end
+    errors.empty?
```

Comment:
Better to use a new life before last return to make it more redable
```suggestion

    errors.empty?
```

Reply:
Fixed in [`f0991a5` (#42)](https://github.com/suho/google-search-ruby/pull/42/commits/f0991a50f82121eb55cc6afc300dc953406ebac0)